packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -151,6 +153,8 @@ export default class CodeGen {
     referencedComponents: Set<string> = new Set();
     apiVersion: APIVersion;
 
+    staticExpressionMap = new WeakMap<Attribute, string>();
```

Comment:
This will eventually expand to include `Text` nodes as well.

Reply:
I don't think this `WeakMap` is strictly necessary. We could just set the property directly on the `Attribute`/`Text` class, since we own it and can do what we want with it.

Reply:
@nolanlawson I tried setting the token directly on the `Attribute`/`Text` but the issue is it will be revealed in the final AST output.

I wrote some logic in the latest 93f51d8c0d62013ab61fb6f499ce79329df14da1 to handle adding / removing the token from the `Attribute`, mostly in the [`static-part-tokens`](https://github.com/salesforce/lwc/pull/4056/commits/93f51d8c0d62013ab61fb6f499ce79329df14da1#diff-096baf6a3d380408ee49b19d6095f1ef99b61e7a86a6dd55c51f1d94f743caca) file.

Let me know what you think, IMO I think as long as we're exposing the AST we should keep it clean (remove the part token before it's exposed).

Alternatively, I placed the token in a few `meta` field, we could just leave a comment saying to ignore everything in that field.

Reply:
I think I was wrong. If the object is exposed externally, then a `WeakMap` is the correct choice!